src/odin_devices/si5324.py

```diff
@@ -0,0 +1,539 @@
+        Since the map is register rather than value-based, there is no need to make use
+        of the _Field access functions.
+
+        :param mapfile_location: location of register map file to be read
+        :param verify: Boolean. If true, read registers back to verify they are written correctly.
+        """
+        f = open(mapfile_location, 'r')
+
+        for line in f.readlines():
+            # The register map starts after general information is printed preceded by '#'
+            if line[0] != '#':
+                # Extract register-value pairing from register map
+                register, value = line.split(',')
+                register = int(register)
+                value = int(value[1:3],16) # Value is in hex
+
+                if register == 136 and (value & 0x40):
+                    logger.info("Ignoring write to iCAL, will be applied next")
+                    continue
+
+                # Write register value
+                logger.info("Writing register {} with value {:02X}".format(register,value))
+                self.write8(register, value)
+
+                if verify:
+                    verify_value = self.readU8(register)
+                    logger.debug("Verifying value written ({:b}) against re-read: {:b}".format(value,verify_value))
+                    if verify_value != value:
+                       raise I2CException(
+                               "Write of byte to register {} failed.".format(register))
+
+        f.close()
+
+        # ICAL-sensitive registers will have been modified during this process
+        self.iCAL_required = True
+        self.calibrate()
+
+    def export_register_map (self, mapfile_location):
+        """
+        Generate a register map file using the current settings in device control
+        registers. This file can then be loaded using apply_register_map(filename).
+
+        :param mapfile_location: location of register map file that will be written to.
+        """
+        f = open(mapfile_location, 'w')
+        f.write("# This register map has been generated for the odin-devices SI5324 driver.\n")
+
+        # The registers that will be read are the ones found in output register
+        # maps from DSPLLsim.
+        for register in SI5324._regmap_registers:
+
+            if register == 136:
+                # This register will read 00, but should be written as 0x40 to match
+                # the versions generated by DSPLLsim. This would trigger an iCAL if
+                # written, but is ignored in apply_register_map().
+                f.write("136, 40h\n")
+                continue
+
+            value = self.readU8(register)
+            logger.info("Read register {}: {:02X}".format(register, value))
+            f.write("{}, {:02X}h\n".format(register, value))
+
+        logger.info("Register map extraction complete, to file: {}".format(mapfile_location))
+        f.truncate()
```

Comment:
Why is the call to `truncate` necessary?

Reply:
The export files from DSPLLsim are larger because they include many comment lines at the start. I wanted to make sure that if one of these files was overwritten, the `truncate()` would cut off any additional lines further than the first 44 I'm currently writing.

Reply:
If you've opened the file with mode `'w'` it will be truncated before writing anyway.

Reply:
Ah I see OK! I will remove it!